src/box/lua/schema.lua

```diff
@@ -328,10 +330,23 @@ local function feedback_save_event(event)
     end
 end
 
-box.begin = function()
+box.begin = function(options)
+    if options then
+        check_param(options, 'options', 'table')
+    end
+    local timeout = options and options["timeout"] or box.cfg.txn_timeout
+    if type(timeout) ~= "number" then
+        box.error(box.error.ILLEGAL_PARAMS, "transaction timeout " ..
+                  "should be of type number")
+    end
+    if timeout <= 0 then
+        box.error(box.error.ILLEGAL_PARAMS, "transaction timeout " ..
+                  "must be greather than 0")
+    end
     if builtin.box_txn_begin() == -1 then
         box.error()
     end
+    builtin.box_txn_set_timeout(timeout)
```

Comment:
Same comment as for tx_process_begin.

Reply:
No! We discussed about it before, by default ALL transactions has timeout == TIMEOUT_INFINITY and box.cfg.txn_timeout does not affect this behaviour! (It's necessary because we don't wan't to affect single statement transactions, applier transactions and so on). We use box_txn_set_timeout to specify timeout in places where we wan't to  specify timeout for transactions: in iproto for streams and in box.begin. So we should call this function with default timeout or with timeout, that was passed to the function.

Reply:
Replied [here](https://github.com/tarantool/tarantool/pull/6473#discussion_r720338637).